src/compiler/transformers/es2015.ts

```diff
@@ -4052,4 +4089,19 @@ namespace ts {
                 };
             })();`
     };
+
+    const templateObjectHelper: EmitHelper = {
+        name: "typescript:makeTemplateObject",
+        scoped: false,
+        priority: 0,
+        text: `
+            var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
+                if (Object.defineProperty) {
+                    return Object.defineProperty(cooked, "raw", { value: raw });

```

Comment:
Would be more readable if both branches shared `return cooked;`:
```ts
if (Object.defineProperty) {
    Object.defineProperty(cooked, "raw", { value: raw });
} else {
    cooked.raw = raw;
}
return cooked;
```

Reply:
Alternatively, you can go for even less readable, and do `return Object.defineProperty ? Object.defineProperty(cooked, "raw", { value: raw }) : (cooked.raw = raw, cooked);`. Can cram the entire helper into one line and it's not bad, actually:
```
var __makeTemplateObject = (this && this.__makeTemplateObject) || (cooked, raw) => Object.defineProperty ? Object.defineProperty(cooked, "raw", { value: raw }) : (cooked.raw = raw, cooked);
```
Too bad we don't transpile helpers 👅

Reply:
Spoke with Wesley; apparently Uglify can do a better job on the following thanks to ternary/comma precedence rules:

```ts
function __makeTemplateObject(cooked, raw) {
    if (Object.defineProperty) {
        Object.defineProperty(cooked, "raw", { value: raw });
    }
    else {
        cooked.raw = raw;
    }
    return cooked;
};
```